### ISSUE TITLE
too many open files

### ISSUE THREAD

**Issue by isuruf** [mamba-org/mamba#554]
On the DTK (development machine for osx-arm64), we are getting the following error with micromamba inside constructor during linking phase.

```
std::__1::system_error: failed to open
/Users/erykoff/miniforge3-test2/pkgs/python-3.8.6-h3098bc4_0_cpython/lib/python3.8/lib-dynload/_testmultiphase.cpython-38-darwin.so:
Too many open files
Miniforge3-4.8.5-5-MacOSX-arm64.sh: line 341: 17030 Abort trap: 6
CONDA_SAFETY_CHECKS=disabled CONDA_EXTRA_SAFETY_CHECKS=no
CONDA_CHANNELS=https://conda-web.anaconda.org/conda-forge
CONDA_PKGS_DIRS="$PREFIX/pkgs" "$CONDA_EXEC" install --offline --file
"$PREFIX/pkgs/env.txt" -yp "$PREFIX"
```

On the DTK, `ulimit -n` is 256. Increasing it makes this operation successful.

cc @erykoff, @beckermr

**Comment by isuruf** [comment]
@wolfv, the new `codesign` operation might be the culprit. Maybe `subprocess::check_call` isn't closing file handles properly

**Comment by wolfv** [comment]
it's possible, I am not very happy with the current subprocess implementation (wanted to change to the one from vcpkg #426 )

**Comment by wolfv** [comment]
There is another source of many open file handles (we open all files we want to download, which sometimes can be many). I have fixed that in #705
